csi: Add GlusterFS directory quota to set capacity limitation for ext…

yugyama opened the pull request:
…ernal gluster volumes

* The provisioner sets GlusterFS directory quota to the subdirectories assigned to PVs on the external GlusterFS server.
* The provisioner connects to the server via ssh by using the secret key and username of the external GlusterFS server in K8s Secret.
* A user creates the K8s Secret during kadalu install to enable this function.

Signed-off-by: Yugo Yamauchi <yugo.yamauchi.cf@hitachi.com>

yugyama committed:
csi: Add GlusterFS directory quota to set capacity limitation for external gluster volumes

* The provisioner sets GlusterFS directory quota to the subdirectories assigned to PVs on the external GlusterFS server.
* The provisioner connects to the server via ssh by using the secret key and username of the external GlusterFS server in K8s Secret.
* A user creates the K8s Secret during kadalu install to enable this function.

the RFC specified @
https://kadalu.io/rfcs/0007-Using-GlusterFS-directory-quota-for-external-gluster-volumes.html

Signed-off-by: Yugo Yamauchi <yugo.yamauchi.cf@hitachi.com>

yugyama committed:
csi: Add GlusterFS directory quota to set capacity limitation for external gluster volumes

* The provisioner sets GlusterFS directory quota to the subdirectories assigned to PVs on the external GlusterFS server.
* The provisioner connects to the server via ssh by using the secret key and username of the external GlusterFS server in K8s Secret.
* A user creates the K8s Secret during kadalu install to enable this function.

Signed-off-by: Yugo Yamauchi <yugo.yamauchi.cf@hitachi.com>

amarts commented on csi/controllerserver.py:
```diff
@@ -197,9 +197,60 @@ def CreateVolume(self, request, context):
+                            "%s" % quota_size
+                        ]
+                        
+                        quota_deem_cmd = [
```
I personally think, 'deem-statfs' option setting should be done just 1 time, so, good to move it to 'mount_glusterfs()' external part.

amarts commented on csi/Dockerfile:
```diff
@@ -7,7 +7,7 @@ RUN python3 -m pip install googleapis-common-protos
 ENV DEBIAN_FRONTEND=noninteractive
 RUN apt-get update -yq && \
-    apt-get install -y --no-install-recommends attr python3 libtirpc3 bash && \
+    apt-get install -y --no-install-recommends attr python3 libtirpc3 bash ssh&& \
```
nitpick: space

amarts reviewed (commented):
Welcome!

Thanks for the PR. Would be great to reference your RFC here (in commit message), so the commit will always have relative documentation.

Just a minor comment from my end. See if it makes sense.

leelavg commented:
Thanks for the feature and documentation too :)

Just a small query, did you get time to run a quick sanity with this featute against devel branch?

Reason being, there were some (internal with less user facing) changes after 0.8.2 release wrt external vols and thought it'd be better to get a confirmation to avoid double spending time testing it again.

vatsa287 committed:
operator: Add volume-id option to storage-config (#567)

This PR enables re-creation of storage-pool by manually entering the volume-id of existing device file or path.
Earlier in case of cleanup of kadalu namespace, pointing to same device/path would throw error.

Fixes: #464
Signed-off-by: Shree Vatsa N <vatsa@kadalu.io>

yugyama committed:
Merge branch 'kadalu:devel' into feature-glusterquota

yugyama committed:
Change quota-deem-statfs command to mount_glusterfs, and minor corrections

Signed-off-by: Yugo Yamauchi <yugo.yamauchi.cf@hitachi.com>

yugyama committed:
Merge branch 'kadalu:devel' into feature-glusterquota

yugyama committed:
Merge remote-tracking branch 'yugyama-fork/feature-glusterquota' into devel-secretquota-fix

yugyama commented:
Thank you for your comments.
I ran the quick test of this implementation with 0.8.2 and the latest kadalu (0.8.3-alpha1).

amarts reviewed (commented):
LGTM

amarts commented:
```
Run make pylint
************* Module controllerserver
csi/controllerserver.py:201:75: C0303: Trailing whitespace (trailing-whitespace)
make: *** [Makefile:78: pylint] Error 16
Error: Process completed with exit code 2.
```

leelavg reviewed (approved):
LGTM

yugyama committed:
Fix: minor change for pylint action

Signed-off-by: Yugo Yamauchi <yugo.yamauchi.cf@hitachi.com>

yugyama committed:
Merge remote-tracking branch 'old-origin/devel' into devel-secretquota-fix

yugyama committed:
Fix: minor change for pylint action csi/volumeutils.py

Signed-off-by: Yugo Yamauchi <yugo.yamauchi.cf@hitachi.com>
